pkg/klusterlet/clusterinfo/distributionInfo_sync.go

```diff
@@ -76,10 +79,15 @@ func (s *distributionInfoSyncer) syncOCPDistributionInfo(ctx context.Context, cl
 		history.Version = historyItem.Version
 		history.Verified = historyItem.Verified
 		ocpDistributionInfo.VersionHistory = append(ocpDistributionInfo.VersionHistory, history)
-		if historyItem.State == "Completed" {
-			ocpDistributionInfo.Version = historyItem.Version
-		}
 	}
+
+	version, err := s.claimLister.Get(clusterclaim.ClaimOpenshiftVersion)
+	if err != nil {
+		klog.Errorf("Failed to get ocp version, error: %v", err)
```

Comment:
should we return the err?

Reply:
> should we return the err?

I do not think it's necessary to return err. As other fields should also be updated.

Reply:
think about when will return an error here? if we can not get claim(except for the not found error), can other places get the claim successfully?
IMO we should return the error if it is not a `NotFound` error.